cpp/src/parquet/file_writer.cc

```diff
@@ -124,10 +142,12 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
 
     ++next_column_index_;
 
-    const ColumnDescriptor* column_descr = col_meta->descr();
-    std::unique_ptr<PageWriter> pager =
-        PageWriter::Open(sink_, properties_->compression(column_descr->path()), col_meta,
-                         properties_->memory_pool());
+    Compression::type compressionCodec;
+    int32_t compression_level;
+    DetermineCompressionCodecAndLevel(properties_, col_meta->descr()->path(),
+                                      compressionCodec, compression_level);
+    std::unique_ptr<PageWriter> pager = PageWriter::Open(
+        sink_, compressionCodec, compression_level, col_meta, properties_->memory_pool());
```

Comment:
This almost suggests it would be better to pass `properties_` and the column path into `PageWriter::Open` instead. Not sure how much refactoring that would be

Reply:
I think it's only the compression type and level which are passed, and in this particular case the memory pool. However, not all three are always passed but rather the compression level and codec. Thus, I do not want to pass properties since there is a lot more unnecessary things there.